components/server/src/authorization/authorizer.ts

```diff
@@ -116,40 +118,61 @@ export class Authorizer {
         );
     }
 
-    async hasPermissionOnProject(userId: string, permission: ProjectPermission, projectId: string): Promise<boolean> {
+    async hasPermissionOnProject(
+        userId: string,
+        permission: ProjectPermission,
+        project: { id: string; teamId?: string },
```

Comment:
lets make this explicit and pass in `projectId: string` and `organizationId: string`